crates/ruff_notebook/src/notebook.rs

```diff
@@ -85,6 +85,23 @@ impl Notebook {
         Self::from_reader(Cursor::new(source_code))
     }
 
+    /// Generate a pseudo-representation of a notebook that can be used
+    /// for linting by the language server. As this is not generated directly from the raw JSON
+    /// of a notebook file, writing this back into the file system is a bad idea.
+    pub fn from_cells(
+        cells: Vec<Cell>,
+        metadata: crate::RawNotebookMetadata,
+    ) -> Result<Self, NotebookError> {
```

Comment:
I'd prefer to move this over in `ruff_server` and expose the `from_raw_notebook` method. This way the separation between the `ruff_server` crate and `ruff_notebook` crate still remains.